core/src/main/scala/org/apache/spark/memory/ExecutionMemoryPool.scala

```diff
@@ -105,11 +106,12 @@ private[memory] class ExecutionMemoryPool(
       lock.notifyAll()
     }
 
+    var waitTimes = 0
     // Keep looping until we're either sure that we don't want to grant this request (because this
     // task would have more than 1 / numActiveTasks of the memory) or we have enough free
     // memory to give it (we always let each task get at least 1 / (2 * numActiveTasks)).
     // TODO: simplify this to limit each task to its own slot
-    while (true) {
+    while (waitTimes < 3) {
```

Comment:
This doesn't look like a general optimization to me, but depends on the specific runtime environment.

Reply:
+1 for @Ngone51 's comment.

Reply:
> This doesn't look like a general optimization to me, but depends on the specific runtime environment.

Yeah, only optimize some special case